Add pulling of screenshots during UI Test for screenshot-tests-for-android

MagicCedric opened the pull request:
Introduce PullScreenshotTestRunListener to pull screenshots from UI-Test if the configuration has a filter that enables it. It will only execute it at the end of the batch of test, and the runner will wait until it finishes (blocking) to finish the batch.

Introduce Configuration.pullScreenshotFilterConfiguration to specify filters that mark which test contains screenshots to PullScreenshotTestRunListener.

Update version to 0.5.3

MagicCedric committed:
Introduce PullScreenshotTestRunListener to pull screenshots from UI-Test if the configuration has a filter that enables it.
Introduce pullScreenshotFilterConfiguration to specify these filters to PullScreenshotTestRunListener.
Add an ability to wait for the pulling of screenshots to end before stopping the work on the batch.
Update version to 0.5.3

lukaville commented on vendor/vendor-android/ddmlib/src/main/kotlin/com/malinskiy/marathon/android/ddmlib/DdmlibAndroidDevice.kt:
```diff
@@ -301,11 +313,18 @@ class DdmlibAndroidDevice(
             prepareRecorderListener(feature, fileManager, devicePoolId, attachmentProviders)
         } ?: NoOpTestRunListener()
 
+        val vrtListener = if (configuration.isPullScreenshotEnabled()) {
```
rename `vrtListener` to `pullScreenshotsListener`?

MagicCedric replied:
Yes, also renamed package

lukaville commented on cli/src/main/kotlin/com/malinskiy/marathon/cli/config/ConfigFactory.kt:
```diff
@@ -67,7 +67,8 @@ class ConfigFactory(private val mapper: ObjectMapper) {
             vendorConfiguration = vendorConfiguration as VendorConfiguration,
-            analyticsTracking = config.analyticsTracking
+            analyticsTracking = config.analyticsTracking,
+            pullScreenshotFilterConfiguration = config.pullScreenshotFilterConfiguration
```
we need to ignore caching for these tests as we do for strict run tests, check `TestCacheLoader:69`. We immediately emit cache miss for strict run tests. Looks like we need to do the same for screenshot tests

MagicCedric replied:
True, done in a similar manner to what we do for strict run

MagicCedric committed:
Rename vrt leftovers to pullScreenshots

MagicCedric committed:
Update cache logic to always run screenshot test and never use the cache results.
We don't save screenshots in cache and cannot be sure that nothing impacted the UI so we always run them.
